projects/packages/sync/src/class-lock.php

```diff
@@ -42,18 +42,37 @@ class Lock {
 	 * @return boolean True if succeeded, false otherwise.
 	 */
 	public function attempt( $name, $expiry = self::LOCK_TRANSIENT_EXPIRY ) {
-		$name        = self::LOCK_PREFIX . $name;
-		$locked_time = get_option( $name );
+
+		global $wpdb;
+		$name = self::LOCK_PREFIX . $name;
+
+		// Options API is explicitly bypassed. This is because with high concurrency request to the
+		// site multiple processes would pass checks and send concurrent data to WP.
+
+		$row         = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", $name ) );
+		$locked_time = false;
+		if ( is_object( $row ) ) {
+			$locked_time = $row->option_value;
+		}
 
 		if ( $locked_time ) {
 			if ( microtime( true ) < $locked_time ) {
 				return false;
+			} else {
+				// If expired delete but don't send. That will occurr in new request to avoid race conditions.
```

Comment:
This doesn't need to be in an else. just close the above if and let this code not need to be indented...

Reply:
:thumbsup:  updated